static/js/compose.js

```diff
@@ -777,14 +777,37 @@ function validate_stream_message() {
 
     return true;
 }
-
+// The function checks whether the recipients are users of the realm or cross realm users (bots for now)
 function validate_private_message() {
     if (exports.recipient() === "") {
         compose_error("Please specify at least one recipient", $("#private_message_recipient"));
         return false;
-    }
+    } else {
+        var private_recipients = util.extract_pm_recipients(compose.recipient());
+        var invalid_recipients = [];
+        _.each(private_recipients, function (email) {
+            // This case occurs when exports.recipient() ends with ','
+            if (email === "") {
+                return;
+            }
+            if (util.string_in_list_case_insensitive(email, page_params.cross_realm_user_emails)) {
+                return;
+            }
+            if (people.realm_get(email) === undefined) {
+                invalid_recipients.push(email);
+            }
```

Comment:
For clarity, I think it'd be better if we did the `people.realm_get` case first, since that's the common case; I'll just reorder that when I merge.